src/plugins/data/public/search/aggs/buckets/lib/time_buckets/time_buckets.ts

```diff
@@ -70,108 +61,17 @@ interface TimeBucketsConfig {
  * @param {[type]} display [description]
  */
 export class TimeBuckets {
-  private getConfig: (key: string) => any;
-
-  private _lb: Bounds['min'] = null;
-  private _ub: Bounds['max'] = null;
+  private _uiSettings: IUiSettingsClient;
```

Comment:
The goal with updating the settings is to make an eventual migration of this code to the server as easy as possible.

`IUiSettingsClient` is different based on whether this runs on the client or server... you can compare both types in `src/core/public/ui_settings/types.ts` and `src/core/server/ui_settings/types.ts` to see what I mean.

Specifically the `get` method returns a promise on the server, but is synchronous on the client. 

This is why in the issue I was suggesting explicitly passing the settings that are needed:
```
constructor({
  'histogram:maxBars': number,
  'histogram:barTarget': number,
  dateFormat: string,
  'dateFormat:scaled': string,
}: TimeBucketsConfig) {}
```

This way, TimeBuckets doesn't need to worry about these differences and can just receive the necessary settings. (On the server we would `await` the relevant ui config settings in our `setup` lifecycle method)

Reply:
done.